model_compression_toolkit/core/pytorch/graph_substitutions/substitutions/functional_batch_norm.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2023 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from torch import nn
+import torch.nn.functional as F
+
+from model_compression_toolkit.core.common.graph.graph_matchers import NodeOperationMatcher
+from model_compression_toolkit.core import common
+from model_compression_toolkit.core.common import BaseNode, Graph
+from model_compression_toolkit.core.pytorch.constants import *
+from model_compression_toolkit.logger import Logger
+
+
+class FunctionalBatchNorm(common.BaseSubstitution):
+    """
+    Replace functional batch_norm with BatchNorm2d.
+    """
+
+    def __init__(self):
+        """
+        Matches: functional batch_norm
+        """
+        bn_node = NodeOperationMatcher(F.batch_norm)
+        super().__init__(matcher_instance=bn_node)
+
+    def get_attributes_from_inputs(self, graph: Graph, node: BaseNode) -> dict:
+        input_nodes = graph.get_prev_nodes(node)
+        if len(input_nodes) == 5:
+            gamma = list(input_nodes[1].weights.values())[0]
+            beta = list(input_nodes[2].weights.values())[0]
+            mean = list(input_nodes[3].weights.values())[0]
+            var = list(input_nodes[4].weights.values())[0]
+        elif len(input_nodes) == 3:
+            mean = list(input_nodes[1].weights.values())[0]
+            var = list(input_nodes[2].weights.values())[0]
+            gamma = None
+            beta = None
+        else:
+            Logger.error(f'functional batch_norm is expected to have 3 or 5 inputs, got {len(input_nodes)}')
+
```

Comment:
Consider to check the tensor dimensions (as you only support banchNorm2d)

Reply:
done